tests/heka/test_message_parser.py

```diff
@@ -4,61 +4,72 @@
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
+import copy
 import json
+import pytest
+from mock import MagicMock
 from moztelemetry.heka import message_parser
 from moztelemetry.util.streaming_gzip import streaming_gzip_wrapper
 
 
-def test_unpack(data_dir):
-    for t in ["plain", "snappy", "mixed", "gzip", "gzip_mixed"]:
-        filename = "{}/test_{}.heka".format(data_dir, t)
-        with open(filename, "rb") as o:
-            if "gzip" in t:
-                o = streaming_gzip_wrapper(o)
-            msg = 0
-            for r, b in message_parser.unpack(o, try_snappy=True):
-                j = json.loads(r.message.payload)
-                assert msg == j["seq"]
-                msg += 1
-            assert 10 == msg
-
+@pytest.mark.parametrize("heka_format", ["plain", "snappy", "mixed", "gzip",
+                                         "gzip_mixed"])
+def test_unpack(data_dir, heka_format):
+    filename = "{}/test_{}.heka".format(data_dir, heka_format)
+    with open(filename, "rb") as o:
+        if "gzip" in heka_format:
+            o = streaming_gzip_wrapper(o)
+        msg = 0
+        for r, b in message_parser.unpack(o, try_snappy=True):
+            j = json.loads(r.message.payload)
+            assert msg == j["seq"]
+            msg += 1
+        assert 10 == msg
 
-def test_unpack_nosnappy(data_dir):
-    expected_counts = {"plain": 10, "snappy": 0, "mixed": 5,
-                       "gzip": 10, "gzip_mixed": 5}
-    for t in expected_counts.keys():
-        count = 0
-        filename = "{}/test_{}.heka".format(data_dir, t)
-        with open(filename, "rb") as o:
-            if "gzip" in t:
-                o = streaming_gzip_wrapper(o)
-            try:
-                for r, b in message_parser.unpack(o, try_snappy=False):
-                    count += 1
-            except:
-                pass
-            assert expected_counts[t] == count
 
+@pytest.mark.parametrize("heka_format,expected_count", [
+    ("plain", 10),
+    ("snappy", 0),
+    ("mixed", 5),
+    ("gzip", 10),
+    ("gzip_mixed", 5)
+])
+def test_unpack_nosnappy(data_dir, heka_format, expected_count):
```

Comment:
Ok, consolidated all of this into a single parametrized test. I think it's easier to see what's going on now...